### PR TITLE
new changes

### PR DESCRIPTION
**Pull request by MounirIssa**
# Abstract

Refactored the code by reducing the number of classes for efficiency.

# Issues

No issue

## Type of Change

Refactored the existing code for better development.

# Testing

No testing

# Checklist

Perform a self-review using the following checklist. Some items may not apply to your PR if it 
is for a bug fix. You can check an item by replacing the [ ] with [x].

- [x] **The code is understandable.**
  - [x] The code is follows formatting conventions (spacing, indents, etc).
  - [x] The code makes use of names that sound like behavior.
  - [x] The code is commented in hard-to-understand areas.
  - [x] The code contains changes to documentation if needed.

- [x] **The code is well structured.**
  - [x] The code is organized into logical components.
  - [x] The components sound like the behavior (requirements) they implement.
  - [x] The components have clearly defined single responsibilities.

- [x] **The code is correct.**
  - [x] The changes include tests that prove the fix is effective or that the feature fully works.
  - [x] The changes generate no new failed tests, build errors, or warnings.

# Pending Items

None.

### CODE REVIEW

File: lib/src/main/java/edu/ucsd/cse110/successorator/lib/domain/TasksRepository.java
Comment (newracket) [inline lib/src/main/java/edu/ucsd/cse110/successorator/lib/domain/TasksRepository.java:1]: Rename to ITasksRepository

File: lib/src/main/java/edu/ucsd/cse110/successorator/lib/domain/Tasks.java
Comment (newracket) [inline lib/src/main/java/edu/ucsd/cse110/successorator/lib/domain/Tasks.java:1]: TasksRepository

File: lib/src/test/java/edu/ucsd/cse110/successorator/lib/domain/FinishedFlashcardRepositoryTest.java
Comment (newracket) [inline lib/src/test/java/edu/ucsd/cse110/successorator/lib/domain/FinishedFlashcardRepositoryTest.java:1]: Bring it back and rename it to TasksRepositoryTest

**Review by newracket — CHANGES_REQUESTED** [review]

**Comment by pink10000** [comment]
so far, lgtm

**Review by newracket — APPROVED** [review]

**Pull request merged**